pkg/archive/archive.go

```diff
@@ -124,15 +125,66 @@ func IsArchivePath(path string) bool {
 	return err == nil
 }
 
+const (
+	ZstdMagicSkippableStart = 0x184D2A50
+	ZstdMagicSkippableMask  = 0xFFFFFFF0
+)
+
+var (
+	Bzip2Magic = []byte{0x42, 0x5A, 0x68}
+	GzipMagic  = []byte{0x1F, 0x8B, 0x08}
+	XzMagic    = []byte{0xFD, 0x37, 0x7A, 0x58, 0x5A, 0x00}
+	ZstdMagic  = []byte{0x28, 0xb5, 0x2f, 0xfd}
+)
+
+type matcher = func([]byte) bool
+
+func magicNumberMatcher(m []byte) matcher {
+	return func(source []byte) bool {
+		return bytes.HasPrefix(source, m)
+	}
+}
+
+// ZstDetector detects zstd compression algorithm.
+// Zstandard compressed data is made of one or more frames.
+// There are two frame formats defined by Zstandard: Zstandard frames and Skippable frames.
+// See https://tools.ietf.org/id/draft-kucherawy-dispatch-zstd-00.html#rfc.section.2 for more details.
+func ZstDetector(buf []byte) bool {
+	if bytes.HasPrefix(buf, ZstdMagic) {
+		// Zstandard frame
+		return true
+	} else {
```

Comment:
Linting failure here, and looks indeed that the `else` is redundant here;

```
pkg/archive/archive.go:156:9: `if` block ends with a `return` statement, so drop this `else` and outdent its block (golint)
	} else {
```

Reply:
Done.